components/CalendarDays.js

```diff
@@ -0,0 +1,116 @@
+import styled from "styled-components";
+import MultipleIcon from "assets/calendarIcons/circles.svg";
+
+export default function CalendarDays({
+  day,
+  getColorByEmotionType,
+  onDayClick,
+  getEmotionsForDay,
+}) {
+  const firstDayOfMonth = new Date(day.getFullYear(), day.getMonth(), 1);
+  const weekdayOfFirstDay = firstDayOfMonth.getDay();
+  let currentDays = [];
+
+  const currentDate = new Date(firstDayOfMonth);
+
+  for (let i = 0; i < 42; i++) {
+    if (i === 0 && weekdayOfFirstDay === 0) {
+      currentDate.setDate(currentDate.getDate() - 7);
+    } else if (i === 0) {
+      currentDate.setDate(currentDate.getDate() + (i - weekdayOfFirstDay));
+    } else {
+      currentDate.setDate(currentDate.getDate() + 1);
+    }
+
+    let calendarDay = {
+      $currentMonth: currentDate.getMonth() === day.getMonth(),
+      date: new Date(currentDate),
+      month: currentDate.getMonth(),
+      number: currentDate.getDate(),
+      $selected: currentDate.toDateString() === day.toDateString(),
+      year: currentDate.getFullYear(),
+    };
+
+    currentDays.push(calendarDay);
+  }
+
+  return (
+    <StyledTableContent>
+      {currentDays.map((calendarDay, index) => (
+        <StyledCalendarDay
+          key={index}
```

Comment:
The key uses the index. While this works, ensure the key is stable to prevent React re-rendering issues when the data changes.

_Example: use the ISO string of the data._

```suggestion
          key={calendarDay.date.toISOString()}
```